src/data/repositories/ProjectD2Repository.ts

```diff
@@ -0,0 +1,110 @@
+import { D2Api } from "$/types/d2-api";
+import { FutureData, apiToFuture } from "$/data/api-futures";
+import { Paginated } from "$/domain/entities/Paginated";
+import { Project } from "$/domain/entities/Project";
+import { GetDataSetOptions } from "$/domain/repositories/DataSetRepository";
+import { ProjectRepository } from "$/domain/repositories/ProjectRepository";
+import _ from "$/domain/entities/generic/Collection";
+import { DataSetD2Api } from "$/data/repositories/DataSetD2Api";
+import { Id } from "$/domain/entities/Ref";
+import { DataSet } from "$/domain/entities/DataSet";
+import { D2CategoryOptionType } from "$/data/repositories/D2ApiCategoryOption";
+import { MetadataItem } from "$/domain/entities/MetadataItem";
+import { Future } from "$/domain/entities/generic/Future";
+
+export class ProjectD2Repository implements ProjectRepository {
+    private d2DataSetApi: DataSetD2Api;
+    constructor(private api: D2Api, private metadata: MetadataItem) {
+        this.d2DataSetApi = new DataSetD2Api(this.api, metadata);
+    }
+
+    getAll(): FutureData<Project[]> {
```

Comment:
As a general rule, be careful with getAll methods that return an array, it's unbounded. There are solutions with Generators, but it's complex. Not saying it has to be changed here, but so we are aware of this.